meteor/sofie-server/data-access/repositories/mongo/mongo-segment-repository.ts

```diff
@@ -33,4 +35,24 @@ export class MongoSegmentRepository extends BaseMongoRepository implements Segme
 			})
 		)
 	}
+
+	public async deleteRundownSegments(rundownId: string): Promise<void> {
+		this.assertDatabaseConnection(this.deleteRundownSegments.name)
+		const segments: Segment[] = await this.getSegments(rundownId)
+
+		segments
+			.map((segment: Segment) => segment.id)
+			.forEach(async (id: string) => this.partRepository.deleteSegmentParts(id))
+		const segmentDeleteResult: DeleteResult = await this.getCollection().deleteMany({ rundownId: rundownId })
+
+		// TODO: Figure out how to archive a 'false' acknowledgment, and add test case using that knowledge
+		if (!segmentDeleteResult.acknowledged) {
+			throw new DeletionFailedException(`Deletion of segments was not acknowledged, for rundownId: ${rundownId}`)
+		}
+		if (segmentDeleteResult.deletedCount === 0) {
+			throw new DeletionFailedException(
+				`Expected to delete one or more segments, but none was deleted, for rundownId: ${rundownId}`
+			)
+		}
```

Comment:
Isn't a rundown with 0 segments technically allowed to exist?

Reply:
Same comment as your similar comment for parts.

For this one i am leaning heavily towards a no however.
What would a rundown with 0 segments do when a take happens?

Reply:
An iNews Queue with 0 stories is valid, so it should also be valid for Rundowns in Sofie to have no Segments. The ingest side supports it, and it would be strange if rundowns started to disappear from the Rundown Overview (Lobby) or if the users were kicked out of the Rundown View when all stories are killed in the rundown they're looking at. Remember that usually there's one _production_ Rundown (Playlist) per gallery, and anything can happen to the stories between shows. You can activate a Rundown with no Segments, but you can't Take until a Segment is added.
![image](https://github.com/tv2/tv-automation-server-core/assets/9103996/784e5fa3-baeb-44fa-a960-3c344e56bf72)

Reply:
@ianshade Do you also know if ingest supports a Segment with no Parts? I would guess that it does. And if it do, then it's a good argument for us also supporting it (atleast, if we are not going to change how ingest works which we would like to avoid at this stage).

Reply:
This sound like the check to ensure that atleast one is deleted, needs to be removed, possibly for all the repositories?

Am i understanding that correctly?

Reply:
Regardless of whether this is supposed to be supported or not, 0 _child_ documents found to be deleted is a potential condition to log a warning at most, but not to throw an exception and prevent a _parent_ document from being deleted. If you throw here, you're likely to leave the users with unwanted data stuck in the database in case some other component of the system failed, e.g. created a rundown but thrown when creating its segments.